test/e2e/ClientOptions.test.js

```diff
@@ -406,48 +419,54 @@ describe('Client console.log', () => {
     },
   ];
 
-  cases.forEach(({ title, options }) => {
-    it(title, (done) => {
-      const res = [];
-      const testOptions = Object.assign({}, baseOptions, options);
-
-      // TODO: use async/await when Node.js v6 support is dropped
-      Promise.resolve()
-        .then(() => {
-          return new Promise((resolve) => {
-            testServer.startAwaitingCompilation(config, testOptions, resolve);
-          });
-        })
-        .then(() => {
-          // make sure the previous Promise is not passing along strange arguments to runBrowser
-          return runBrowser();
-        })
-        .then(({ page, browser }) => {
-          return new Promise((resolve) => {
-            page.goto(`http://localhost:${port2}/main`);
-            page.on('console', ({ _text }) => {
-              res.push(_text);
+  transportModes.forEach((mode) => {
+    cases.forEach(({ title, options }) => {
+      title += ` (${
+        Object.keys(mode).length ? mode.transportMode : 'default'
+      })`;
+      options = { ...mode, ...options };
+      it(title, (done) => {
+        const res = [];
+        const testOptions = Object.assign({}, baseOptions, options);
+
+        // TODO: use async/await when Node.js v6 support is dropped
```

Comment:
You can use async/await already.

Reply:
@hiroppy I tried many times to use async/await but every time I get errors or wrong results, my last attempt was : 
```js
cases.forEach(async ({ title, options }) => {
      title += ` (${
        Object.keys(mode).length ? mode.transportMode : 'default'
      })`;
      options = { ...mode, ...options };
      testOptions = Object.assign({}, baseOptions, options);
      await testServer.startAwaitingCompilation(config, testOptions); // and without await
      it(title, (done) => {

        const res = [];
          runBrowser().then(({ page, browser }) => {
            page.goto(`http://localhost:${port2}/main`);
            page.on('console', ({ _text }) => {
              res.push(_text);
            });

            // wait for load before closing the browser
            page.waitForNavigation({ waitUntil: 'load' }).then(() => {
              page.waitFor(beforeBrowserCloseDelay).then(() => {
                browser.close().then(() => {
                  // Order doesn't matter, maybe we should improve that in future
                  expect(res.sort()).toMatchSnapshot();
                  testServer.close(done)
                });
              });
            });
          });
      });
    });
```
for now It works fine, I need help with rewriting this with async/await. 
@Loonride any thoughts for this ?

Reply:
I believe the issue is forEach won't work with async, at the very top where you do `cases.forEach`. Try doing a normal for loop and see if that fixes it

Reply:
@Loonride thanks for the hint, what fixed the issue was adding `await` before `it` & adding the server starter inside `it`, not sure if converting the other promises helped ex : 
```
await it(title, async (done) => {
        await testServer.startAwaitingCompilation(config, testOptions);
        const res = [];
        const { page, browser } = await runBrowser();
        page.goto(`http://localhost:${port2}/main`);
        page.on('console', ({ _text }) => {
          res.push(_text);
        });
        // wait for load before closing the browser
        await page.waitForNavigation({ waitUntil: 'load' });
        await page.waitFor(beforeBrowserCloseDelay);
        await browser.close();
        // Order doesn't matter, maybe we should improve that in future
        await expect(res.sort()).toMatchSnapshot();
        await testServer.close(done);
      });
```
It's strange that using `for` for the inner loop didn't work & skipped many tests.